### PR TITLE
#3 exception handling in controller

### PR DESCRIPTION
**Pull request by ZsoltToth**

### CODE REVIEW

File: controller/src/main/java/uni/miskolc/ips/ilona/measurement/controller/MeasurementController.java [hunk 1]
```diff
@@ -131,17 +119,7 @@ public final boolean recordMeasurement(@RequestBody final MeasurementRegistratio
         if (measurementRegistrationRequest.getRfidtags() != null) {
             measurement.setRfidtags(new RFIDTags(new HashSet<byte[]>(measurementRegistrationRequest.getRfidtags().getRfidTag())));
```
Comment (judit0310) [inline controller/src/main/java/uni/miskolc/ips/ilona/measurement/controller/MeasurementController.java]: In previous if-s, the creation of the instance is not the parameter of a setter.

File: controller/src/main/java/uni/miskolc/ips/ilona/measurement/controller/GlobalControllerExceptionHandler.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+ * Global exception handler for controllers.
+ *
+ * If a database is not available the services should return with Insufficient storege status code (507).
+ *
+ * If there is a conversion problem during the marshalling of JSON messages then controllers return with unsupported media type satus code (415).
+ *
+ * @author tothzs
+ */
+@ControllerAdvice
+public class GlobalControllerExceptionHandler {
```
Comment (judit0310) [inline controller/src/main/java/uni/miskolc/ips/ilona/measurement/controller/GlobalControllerExceptionHandler.java]: These are not logged.

File: controller/src/main/java/uni/miskolc/ips/ilona/measurement/controller/ZoneController.java [hunk 3]
```diff
@@ -40,67 +41,37 @@ public ZoneController(ZoneService zoneManagerService) {
...
 /**
  * Adds a new zone to the list of zones.
  * @param name The name of the new zone
- * @return Returns true if successful, or an exception if not.
  */
 	@RequestMapping("/addZone")
-	public @ResponseBody boolean addZone(@RequestParam("name")final String name) {
-		//TODO return with void
+	@ResponseBody
+	public void addZone(@RequestParam("name")final String name) throws ZoneNotFoundException, DatabaseUnavailableException {
```
Comment (judit0310) [inline controller/src/main/java/uni/miskolc/ips/ilona/measurement/controller/ZoneController.java]: Why can it throw ZoneNotFoundException?

File: controller/src/main/java/uni/miskolc/ips/ilona/measurement/controller/MeasurementController.java (hunk 1 above)
Comment (ZsoltToth) [reply controller/src/main/java/uni/miskolc/ips/ilona/measurement/controller/MeasurementController.java]: instantiation has been moved before the setter method

**Comment by ZsoltToth** [comment]
Change requests are fulfilled.

**Pull request merged**